### ISSUE TITLE
mini.ai newline bug

### ISSUE THREAD

**Issue by simonmandlik** [echasnovski/mini.nvim#902]
### Contributing guidelines

- [X] I have read [CONTRIBUTING.md](https://github.com/echasnovski/mini.nvim/blob/main/CONTRIBUTING.md)
- [X] I have read [CODE_OF_CONDUCT.md](https://github.com/echasnovski/mini.nvim/blob/main/CODE_OF_CONDUCT.md)
- [X] I have updated 'mini.nvim' to latest version

### Module(s)

mini.ai

### Description

I'm using default settings, search method is thus `cover_or_next`.

```
[<cursor>,
 b, [c] ]
```

Placing the cursor anywhere in `<cursor>`, `di]` correctly deletes everything inside outer square brackets.

However, in

```
[<cursor>, b, [c]
  ]
```

`di]` deletes only the `c`.

In

```
[<cursor>, b,
  [c]
  ]
```

I get the correct behavior again.

### Neovim version

0.10.0

### Steps to reproduce

`repro.lua`:

### Actual behavior

```lua
local root = vim.fn.fnamemodify("./.repro", ":p")
for _, name in ipairs({ "config", "data", "state", "cache" }) do
    vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end
local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
    vim.fn.system({ "git", "clone", "--filter=blob:none", "https://github.com/folke/lazy.nvim.git", lazypath, })
end
vim.opt.runtimepath:prepend(lazypath)

local plugins = {
    {
        {
            "echasnovski/mini.ai",
            opts = {},
        },
    }
}

require("lazy").setup(plugins, {
    root = root .. "/plugins",
})
```

**Comment by echasnovski** [comment]
Thanks for the issue!

This is an intended behavior of [first searching only inside current line/selection](https://github.com/echasnovski/mini.nvim/blob/e94945129780fa5b43447ecc0a4b0273892043e3/doc/mini-ai.txt#L424-L427) before searching in the whole neighborhood. See [this comment](https://github.com/echasnovski/mini.nvim/issues/790#issuecomment-2045602383) for a suggested workaround.

Closing as duplicate of #790, #392, #123, #116.

**Comment by simonmandlik** [comment]
@echasnovski 

First off, apologies for failing the search for duplicates 🫣

Would you perhaps be willing to revisit https://github.com/echasnovski/mini.nvim/issues/790#issuecomment-2083117530? I try to keep the neovim behavior as close as possible to `nvim --clean`, only enhancing it, not modifying. In vanilla neovim, all examples I showed above end up with `[]`. And I really like this modus operandi — perform action at the cursor location and if not possible, move forward. This is true for not only text objects, but also e.g. for <C-a/x>, and many surround plugins.

I know there are `n` and `l` modifiers and in my opinion it is perfectly valid and concise configuration as well, it is just not consistent with the original neovim behavior. In the ideal world, I would like `mini.ai` to behave as original neovim, provide some additional text objects, and also provided `n` and `l` in cases I really need it.

I really like everything else `mini.ai` brings to the table, but not being able to emulate original neovim behavior in this case is a big con for me.

**Comment by echasnovski** [comment]
> Would you perhaps be willing to revisit [#790 (comment)](https://github.com/echasnovski/mini.nvim/issues/790#issuecomment-2083117530)?

Sorry, but nothing has changed in 3 weeks and [this](https://github.com/echasnovski/mini.nvim/issues/790#issuecomment-2083132946) is still the suggested approach. To the extent that if I'd to implement 'mini.ai' now from scratch, it will probably be the default without all 'cover_or_xxx' search method.

**Comment by simonmandlik** [comment]
Yes, I've seen the suggested approach. But `search_mode = "cover"` means:

- give up forward search completely, or
- learn to press one more key in this situation, which as I explained is a different mental model from vanilla neovim and even other mainstream plugins, e.g. `nvim-surround`.

`mini.ai` is an amazing plugin and thanks a lot for it, it's just a pity that migrating to it involves having to rewire you muscle memory.
